doc/generate-subcodes-doc.js

```diff
@@ -50,7 +50,7 @@ function buildSubcodesDoc(errorCodesFiles) {
         const code = domain.code << 24
           | subdomain.code << 16
           | error.code;
-        doc += `\`${code}\`  | \`${error.message.replace(/%s/g, '<placeholder>')}\` | [${error.class}](https://docs.kuzzle.io/core/1/api/essentials/errors/#${error.class.toLowerCase()}) | ${errorName} | ${domainName}.${subdomainName}.${errorName}\n`;
+        doc += `\`0x${code.toString(16)}\`  | \`${error.message.replace(/%s/g, '<placeholder>')}\` | [${error.class}](https://docs.kuzzle.io/core/1/api/essentials/errors/#${error.class.toLowerCase()}) | ${errorName} | ${domainName}.${subdomainName}.${errorName}\n`;
```

Comment:
To make our error codes readable, understandable and predictable for our users, we need an exact hexadecimal representation of a 32 bits integer, meaning that it should always have a fixed length of 8:

```
0xXX YY ZZZZ
  ^  ^  ^
  |  |  error code
  |  subdomain code
  domain code
```

Only converting a number to a string will discard non-significant zeroes.

For instance, this code: 0, 1, 2
Should print: `0x00010002`
This version prints: `0x1002` instead

You should allocate a fixed-length buffer at the start of your fuction: `const buffer = Buffer.allocUnsafe(4);`

And use it like that:

```
        const code = domain.code << 24
          | subdomain.code << 16
          | error.code;

        buffer.writeUInt32BE(code, 0);

        doc += `\`0x${buffer.toString('hex')}\`  | \`${error.message.replace(/%s/g, '<placeholder>')}\` | [${error.class}](https://docs.kuzzle.io/core/1/api/essentials/errors/#${error.class.toLowerCase()}) | ${errorName} | ${domainName}.${subdomainName}.${errorName}\n`;
```

Reply:
Thanks for the explanation